src/VirtoCommerce.MarketingModule.Data/Services/CouponService.cs

```diff
@@ -67,6 +66,13 @@ public async Task SaveCouponsAsync(Coupon[] coupons)
                         }
                         else
                         {
+                            var couponWithSameName = repository.Coupons
+                                .Include(x => x.Promotion)
+                                .FirstOrDefault(x => x.Code == coupon.Code && x.PromotionId == coupon.PromotionId);
+                            if (couponWithSameName != null)
+                            {
+                                throw new ArgumentException($"Coupon with Name: '{couponWithSameName.Code}' for Promotion: '{couponWithSameName.Promotion.Name}' already exist");
```

Comment:
Better throw `InvalidOperationException($"Coupon with Name: '{couponWithSameName.Code}' for Promotion: '{couponWithSameName.Promotion.Name}' already exists");`

Reply:
fixed